src/Public/OpenApi.ps1

```diff
@@ -9,105 +9,410 @@ Enables the OpenAPI default route in Pode, as well as setting up details like Ti
+
 .EXAMPLE
 Enable-PodeOpenApi -Title 'My API' -Version '1.0.0' -RouteFilter '/api/*'
 
 .EXAMPLE
 Enable-PodeOpenApi -Title 'My API' -Version '1.0.0' -RouteFilter '/api/*' -RestrictRoutes
 
 .EXAMPLE
-Enable-PodeOpenApi -Path '/docs/openapi' -Title 'My API' -Version '1.0.0'
+Enable-PodeOpenApi -Path '/docs/openapi'   -NoCompress -Mode 'Donwload' -DisableMinimalDefinitions
 #>
 function Enable-PodeOpenApi {
     [CmdletBinding()]
     param(
-        [Parameter()]
         [ValidateNotNullOrEmpty()]
         [string]
         $Path = '/openapi',
 
-        [Parameter(Mandatory = $true)]
+        [Parameter(ParameterSetName = 'Deprecated')]
         [string]
         $Title,
 
-        [Parameter()]
-        [ValidateNotNullOrEmpty()]
+        [Parameter(ParameterSetName = 'Deprecated')]
+        [ValidatePattern('^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$')]
         [string]
-        $Version = '0.0.0',
+        $Version,
 
-        [Parameter()]
+        [Parameter(ParameterSetName = 'Deprecated')]
         [string]
         $Description,
 
-        [Parameter()]
+        [ValidateSet('3.1.0', '3.0.3', '3.0.2', '3.0.1', '3.0.0')]
+        [string]
+        $OpenApiVersion = '3.0.3',
+
         [ValidateNotNullOrEmpty()]
         [string]
         $RouteFilter = '/*',
 
-        [Parameter()]
+        [string[]]
+        $EndpointName,
+
         [object[]]
         $Middleware,
 
+        [Parameter()]
+        [Alias('Auth')]
+        [string]
+        $Authentication,
+
+        [Parameter()]
+        [string[]]
+        $Role,
+
+        [Parameter()]
+        [string[]]
+        $Group,
+
+        [Parameter()]
+        [string[]]
+        $Scope,
+
+        [switch]
+        $RestrictRoutes,
+
+        [ValidateSet('View', 'Download')]
+        [String]
+        $Mode = 'view',
+
+        [ValidateSet('Json', 'Json-Compress', 'Yaml')]
+        [String]
+        $MarkupLanguage = 'Json',
+
+        [switch]
+        $EnableSchemaValidation,
+
+        [ValidateRange(1, 100)]
+        [int]
+        $Depth = 20,
+
+        [switch]
+        $DisableMinimalDefinitions,
+
+        [Parameter(Mandatory, ParameterSetName = 'DefaultResponses')]
+        [hashtable]
+        $DefaultResponses,
+
+        [Parameter(Mandatory, ParameterSetName = 'NoDefaultResponses')]
         [switch]
-        $RestrictRoutes
+        $NoDefaultResponses,
+
+        [string]
+        $DefinitionTag
+
     )
+    if (Test-PodeIsEmpty -Value $DefinitionTag) {
+        $DefinitionTag = $PodeContext.Server.OpenAPI.SelectedDefinitionTag
+    }
+    if ($Description -or $Version -or $Title) {
+        if (! $Version) {
+            $Version = '0.0.0'
+        }
+        Write-PodeHost -ForegroundColor Yellow "WARNING: The parameter Title,Version and Description are deprecated. Please use 'Add-PodeOAInfo' instead."
```

Comment:
From testing, so it's a little more obvious in the console, could we tweak the message to something like the following so it includes the function:
```powershell
Write-PodeHost -ForegroundColor Yellow "WARNING: Title, Version, and Description on 'Enable-PodeOpenApi' are deprecated. Please use 'Add-PodeOAInfo' instead."
```